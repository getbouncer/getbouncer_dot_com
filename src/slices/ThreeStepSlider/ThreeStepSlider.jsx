import React, { useState } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Container from '~globals/Container'
import arrowLeft from '~images/arrow-thin-left.svg'
import arrowRight from '~images/arrow-thin-right.svg'
import arrowStep from '~images/arrow-step.svg'
import { AboveSmartphone, Smartphone } from '~components/Media'
import {
  Content,
  Count,
  Controls,
  Arrow,
  ArrowStepOne,
  ArrowStepTwo,
  FlippedArrow,
  Headline,
  Image,
  ImageItem,
  ImagesWrapper,
  ImageTitle,
  Step,
  Text,
  TextWrapper,
  StepWrapper,
  Title,
  Wrapper,
} from './styles'

const ThreeStepSlider = ({ input }) => {
  const [activeIndex, setActiveIndex] = useState(0)
  const { items } = input

  const arrowOneClass = classNames(
    activeIndex >= 1 ? 'active' : null,
    activeIndex === items.length - 1 ? 'flipped' : null,
  )

  const arrowTwoClass = classNames(activeIndex > 1 ? 'active' : null)

  return (
    <Wrapper>
      <Container>
        <Headline>
          <strong>How it works</strong>
        </Headline>
        <Content>
          <TextWrapper>
            <Step>
              {items.map((step, index) => (
                <StepWrapper key={step.step_title} className={index === activeIndex ? 'active' : undefined}>
                  <Title>{step.step_title}</Title>
                  <Text>{step.step_description.text}</Text>
                </StepWrapper>
              ))}
              <Count>
                <span>{activeIndex + 1}</span>
              </Count>
              <AboveSmartphone>
                <Controls>
                  <Arrow
                    className={activeIndex === 0 ? 'disabled' : undefined}
                    onClick={() => setActiveIndex(activeIndex - 1)}
                  >
                    <img src={arrowLeft} alt="Arrow left" />
                  </Arrow>
                  <Arrow
                    className={activeIndex === items.length - 1 ? 'disabled' : undefined}
                    onClick={() => setActiveIndex(activeIndex + 1)}
                  >
                    <img src={arrowRight} alt="Arrow right" />
                  </Arrow>
                </Controls>
              </AboveSmartphone>
            </Step>
          </TextWrapper>
          <ImagesWrapper>
            {items.map((step, index) => (
              <ImageItem key={step.illustration_title} className={index <= activeIndex ? 'active' : undefined}>
                <ImageTitle>{step.illustration_title}</ImageTitle>
                <Image src={step.step_illustration.url} alt={step.step_illustration.alt} />
              </ImageItem>
            ))}
            <ArrowStepOne className={arrowOneClass}>
              <img src={arrowStep} alt="Arrow" />
            </ArrowStepOne>
            <ArrowStepTwo className={arrowTwoClass}>
              <span>1 second</span>
              <FlippedArrow src={arrowStep} alt="Arrow" />
              <img src={arrowStep} alt="Arrow" />
            </ArrowStepTwo>
          </ImagesWrapper>
          <Smartphone>
            <Controls>
              <Arrow
                className={activeIndex === 0 ? 'disabled' : undefined}
                onClick={() => setActiveIndex(activeIndex - 1)}
              >
                <img src={arrowLeft} alt="Arrow left" />
              </Arrow>
              <Arrow
                className={activeIndex === items.length - 1 ? 'disabled' : undefined}
                onClick={() => setActiveIndex(activeIndex + 1)}
              >
                <img src={arrowRight} alt="Arrow right" />
              </Arrow>
            </Controls>
          </Smartphone>
        </Content>
      </Container>
    </Wrapper>
  )
}

ThreeStepSlider.propTypes = {
  input: PropTypes.shape().isRequired,
}

export default ThreeStepSlider
