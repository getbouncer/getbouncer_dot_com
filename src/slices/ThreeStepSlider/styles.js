import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const Wrapper = styled.section`
  padding: 100px 0 200px;
  background-image: linear-gradient(180deg, #ffffff 0%, #d4dafc 100%);

  @media ${breakpoints.smartphone} {
    padding: 50px 0;
  }
`

export const Headline = styled.h2`
  text-align: center;
  margin-bottom: 90px;
  font-size: 2.857rem;

  strong {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
`

export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  width: calc(100% + 50px);

  @media ${breakpoints.smartphone} {
    display: block;
    width: 100%;
  }
`

export const TextWrapper = styled.div`
  display: block;
  width: 365px;

  @media ${breakpoints.smartphone} {
    width: 100%;
  }
`

export const ImagesWrapper = styled.div`
  display: flex;
  max-width: 850px;
  margin-left: 90px;
  position: relative;

  @media ${breakpoints.smartphone} {
    justify-content: space-between;
    margin-left: 0;
    margin-top: 320px;
  }
`

export const Count = styled.div`
  align-items: center;
  background: ${theme.colors.blue};
  border-radius: 100%;
  color: ${theme.colors.white};
  display: flex;
  font-size: 22px;
  font-weight: 600;
  height: 40px;
  justify-content: center;
  width: 40px;

  span {
    position: relative;
    top: -2px;
  }
`

export const Title = styled.h3`
  display: block;
  margin: 26px 3px 22px 0;
`

export const Text = styled.p`
  display: block;
`

export const Step = styled.div`
  display: block;
  position: relative;

  @media ${breakpoints.smartphone} {
    width: 100%;
  }
`

export const StepWrapper = styled.div`
  height: 280px;
  opacity: 0;
  pointer-events: none;
  position: absolute;
  top: 40px;
  width: 365px;

  @media ${breakpoints.smartphone} {
    width: 100%;
  }

  &.active {
    opacity: 1;
    pointer-events: all;
  }
`

export const ImageItem = styled.div`
  display: block;
  opacity: 0.3;
  transition: opacity 0.3s ease-in-out;

  &.active {
    opacity: 1;
  }

  &:not(:last-child) {
    margin-right: 15%;

    @media ${breakpoints.smartphone} {
      margin-right: 5%;
    }
  }
`

export const ImageTitle = styled.p`
  color: ${theme.colors.blue};
  display: block;
  font-size: 22px;
  line-height: 24px;
  margin: 0 auto 15px;
  max-width: 237px;
  text-align: center;

  @media ${breakpoints.smartphone} {
    font-size: 16px;
    font-weight: 600;
    line-height: 20px;
  }
`

export const Image = styled.img`
  display: block;
`

export const Controls = styled.div`
  display: block;
  margin-top: 290px;

  @media ${breakpoints.smartphone} {
    display: flex;
    justify-content: center;
    margin-top: 40px;
  }
`

export const Arrow = styled.div`
  background: ${theme.colors.blue};
  border-radius: 100%;
  cursor: pointer;
  display: block;
  display: inline-block;
  height: 36px;
  margin-right: 22px;
  position: relative;
  transition: opacity 0.2s;
  width: 36px;

  img {
    left: 8px;
    position: absolute;
    top: 10px;
    width: 17px;
  }

  &.disabled {
    opacity: 0.5;
    pointer-events: none;
  }
`

export const ArrowStepOne = styled.div`
  left: 28%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  transition: opacity 0.3s ease-in-out, transform 0s ease 0.6s;
  opacity: 0;

  &.active {
    opacity: 1;
  }

  &.flipped {
    transform: translate(-50%, -50%) rotate(180deg);
  }

  @media ${breakpoints.smartphone} {
    left: 32%;
    top: 60%;

    img {
      width: 50px;
    }
  }
`

export const ArrowStepTwo = styled.div`
  left: 54%;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);

  span,
  img {
    display: block;
    opacity: 0;
  }

  span {
    color: ${theme.colors.blue};
    text-align: center;
    transition: opacity 0.2s ease-in-out 0.4s;
  }

  &.active {
    span,
    img {
      opacity: 1;
    }
  }

  @media ${breakpoints.smartphone} {
    left: 60%;
    top: 60%;

    img {
      width: 50px;
    }

    span {
      font-size: 12px;
    }
  }
`

export const FlippedArrow = styled.img`
  margin-bottom: 15px;
  transform: rotate(180deg);
  transition: opacity 0.2s ease-in-out 0.4s;
`
