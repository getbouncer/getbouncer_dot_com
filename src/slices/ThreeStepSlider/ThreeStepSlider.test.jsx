import React from 'react'
import renderer from 'react-test-renderer'
import ThreeStepSlider from './ThreeStepSlider'

/* eslint-disable no-undef */

describe('ThreeStepSlider', () => {
  it('renders correctly', () => {
    const data = {
      items: [
        {
          step_title: 'step 1 title',
          step_description: {
            text: 'step 1 description',
          },
          illustration_title: 'illustration 1 title',
          step_illustration: {
            alt: 'step 1 illustration',
            url: 'https://www.example.com/step_1_illustration.png',
          },
        },
        {
          step_title: 'step 2 title',
          step_description: {
            text: 'step 2 description',
          },
          illustration_title: 'illustration 2 title',
          step_illustration: {
            alt: 'step 2 illustration',
            url: 'https://www.example.com/step_2_illustration.png',
          },
        },
      ],
    }
    const tree = renderer
      .create(<ThreeStepSlider input={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
