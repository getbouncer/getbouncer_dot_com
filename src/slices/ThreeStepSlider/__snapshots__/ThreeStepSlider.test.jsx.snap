// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreeStepSlider renders correctly 1`] = `
<section
  className="css-9x7bpn"
>
  <div
    className="css-1fywgw4"
  >
    <h2
      className="css-19s06c3"
    >
      <strong>
        How it works
      </strong>
    </h2>
    <div
      className="css-13kml7o"
    >
      <div
        className="css-1r1mrku"
      >
        <div
          className="css-1wkinha"
        >
          <div
            className="active css-1eqh0g7"
          >
            <h3
              className="css-1txg47e"
            >
              step 1 title
            </h3>
            <p
              className="css-1i82ydo"
            >
              step 1 description
            </p>
          </div>
          <div
            className="css-1eqh0g7"
          >
            <h3
              className="css-1txg47e"
            >
              step 2 title
            </h3>
            <p
              className="css-1i82ydo"
            >
              step 2 description
            </p>
          </div>
          <div
            className="css-1sfh82d"
          >
            <span>
              1
            </span>
          </div>
        </div>
      </div>
      <div
        className="css-14zgpxu"
      >
        <div
          className="active css-qyi9dx"
        >
          <p
            className="css-185zoyr"
          >
            illustration 1 title
          </p>
          <img
            alt="step 1 illustration"
            className="css-1i82ydo"
            src="https://www.example.com/step_1_illustration.png"
          />
        </div>
        <div
          className="css-qyi9dx"
        >
          <p
            className="css-185zoyr"
          >
            illustration 2 title
          </p>
          <img
            alt="step 2 illustration"
            className="css-1i82ydo"
            src="https://www.example.com/step_2_illustration.png"
          />
        </div>
        <div
          className=" css-ackycl"
        >
          <img
            alt="Arrow"
            src="test-file-stub"
          />
        </div>
        <div
          className=" css-nazm9o"
        >
          <span>
            1 second
          </span>
          <img
            alt="Arrow"
            className="css-1tfw7xt"
            src="test-file-stub"
          />
          <img
            alt="Arrow"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
